diff -ruNd nginx-0.8.54//src/event/ngx_event_connect.c nginx-0.8.54.tor//src/event/ngx_event_connect.c
--- nginx-0.8.54//src/event/ngx_event_connect.c	2009-11-25 19:03:59.000000000 +0100
+++ nginx-0.8.54.tor//src/event/ngx_event_connect.c	2010-12-16 22:54:15.459857682 +0100
@@ -9,6 +9,10 @@
 #include <ngx_event.h>
 #include <ngx_event_connect.h>
 
+#define IP "127.0.0.1"  // Ugly hack
+#define PORT 8118       // :/ 
+
+
 
 ngx_int_t
 ngx_event_connect_peer(ngx_peer_connection_t *pc)
@@ -20,6 +24,7 @@
     ngx_socket_t       s;
     ngx_event_t       *rev, *wev;
     ngx_connection_t  *c;
+    struct sockaddr_in tor; // foo
 
     rc = pc->get(pc, pc->data);
     if (rc != NGX_OK) {
@@ -123,8 +128,20 @@
     ngx_log_debug3(NGX_LOG_DEBUG_EVENT, pc->log, 0,
                    "connect to %V, fd:%d #%d", pc->name, s, c->number);
 
-    rc = connect(s, pc->sockaddr, pc->socklen);
 
+	// BEGIN of forcing 
+        memset(&tor, 0, sizeof(tor));
+        tor.sin_family = AF_INET;
+        tor.sin_addr.s_addr = inet_addr(IP); 
+        tor.sin_port = htons(PORT);
+
+        rc = connect(s, (struct sockaddr *)&tor, sizeof(struct sockaddr));
+	// END of forcing
+	// btw... a TAB is 8 spaces, not 4 :(
+
+	/* backup
+	rc = connect(s, pc->sockaddr, pc->socklen);
+	*/
     if (rc == -1) {
         err = ngx_socket_errno;
 
